File: Decorator/Java/ConceptSample/Decorator.java

public abstract class Decorator extends Component {
    private Component component = null;
    //通過構造函數傳遞被修飾者
    public Decorator(Component _component) {
        this.component = _component;
    }
    //委托給被修飾者執行
    @Override
    public void operate() {
        this.component.operate();
    }
}

File: Decorator/PHP/DateDecorator (MultipleCompoment)/Decorator.php
<?php

include_once ('IComponent.php');

// Decorator 參與者
abstract class Decorator extends IComponent
{

    public function getAge(): int
    {
        return $this->ageGroup;
    }

    public function setAge(int $ageNow)
    {
        $this->ageGroup = $ageNow;
    }
}